<img src = "https://biblioteca.aurens.com/colleges/logos/17.png" width = 400>


<h1 align=center><font size = 5>From Requirements to Collection</font></h1>


## Introduction



## Objectives

After complting this lab you will be able to:

-   Understand Data Requirements
-   Explore the stages in Data Collection


## Table of Contents

<div class="alert alert-block alert-info" style="margin-top: 20px">
    
1. [Data Requirements](#0)<br>
2. [Data Collection](#2)<br>
</div>
<hr>


# Data Requirements <a id="0"></a>


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig1_flowchart_data_requirements.png" width=500>


In the videos, we learned that the chosen analytic approach determines the data requirements. Specifically, the analytic methods to be used require certain data content, formats and representations, guided by domain knowledge.


In the **From Problem to Approach Lab**, we determined that automating the process of determining the cuisine of a given recipe or dish is potentially possible using the ingredients of the recipe or the dish. In order to build a model, we need extensive data of different cuisines and recipes.


Identifying the required data fulfills the data requirements stage of the data science methodology.

* * *


# Data Collection <a id="2"></a>


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig2_flowchart_data_collection.png" width=500>


In the initial data collection stage, data scientists identify and gather the available data resources. These can be in the form of structured, unstructured, and even semi-structured data relevant to the problem domain.


#### Web Scraping of Online Food Recipes

A researcher named Yong-Yeol Ahn scraped tens of thousands of food recipes (cuisines and ingredients) from three different websites, namely:


<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig3_allrecipes.png" width=500>
<div align="center">
www.allrecipes.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig4_epicurious.png" width=500>
<div align="center">
www.epicurious.com
</div>
<br/><br/>
<img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/images/lab2_fig5_menupan.png" width=500>
<div align="center">
www.menupan.com
</div>
<br/><br/>


For more information on Yong-Yeol Ahn and his research, you can read his paper on [Flavor Network and the Principles of Food Pairing](http://yongyeol.com/papers/ahn-flavornet-2011.pdf?utm_email=Email&utm_source=Nurture&utm_content=000026UJ&utm_term=10006555&utm_campaign=PLACEHOLDER&utm_id=SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987).


Luckily, we will not need to carry out any data collection as the data that we need to meet the goal defined in the business understanding stage is readily available.


### Using this notebook:

To run any of the following cells of code, you can type **Shift + Enter** to excute the code in a cell.


Get the version of R installed.


In [5]:
# check R version
R.Version()$version.string

[1] "R version 3.6.1 (2019-07-05)"

Download the data from the IBM server.


In [20]:
# click here and press Shift + Enter
download.file("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork/labs/Module%202/recipes.csv",
              destfile = "C:\ Users\ Luis\ JupiterNotebooks\ recipes.csv", quiet = TRUE)

print("Done!") # takes about 30 seconds

[1] "Done!"


Read the data into an R dataframe and name it **recipes**.


In [25]:
recipes <- read.csv("C:\ Users\ Luis\ JupiterNotebooks\ recipes.csv") # takes 10 sec

Show the first few rows.


In [26]:
head(recipes)

country    almond angelica anise anise_seed apple apple_brandy apricot
1 Vietnamese No     No       No    No         No    No           No     
2 Vietnamese No     No       No    No         No    No           No     
3 Vietnamese No     No       No    No         No    No           No     
4 Vietnamese No     No       No    No         No    No           No     
5 Vietnamese No     No       No    No         No    No           No     
6 Vietnamese No     No       No    No         No    No           No     
  armagnac artemisia ... whiskey white_bread white_wine whole_grain_wheat_flour
1 No       No        ... No      No          No         No                     
2 No       No        ... No      No          No         No                     
3 No       No        ... No      No          No         No                     
4 No       No        ... No      No          No         No                     
5 No       No        ... No      No          No         No                     
6 No       No        ... No      No          No         No                     
  wine wood yam yeast yogurt zucchini
1 No   No   No  No    No     No      
2 No   No   No  No    No     No      
3 No   No   No  No    No     No      
4 No   No   No  No    No     No      
5 No   No   No  No    No     No      
6 No   No   No  No    No     No

Get the dimensions of the dataframe.


In [27]:
nrow(recipes)

[1] 57691

In [28]:
ncol(recipes)

[1] 384

So our dataset consists of 57,691 recipes. Each row represents a recipe, and for each recipe, the corresponding cuisine is documented as well as whether 384 ingredients exist in the recipe or not, beginning with almond and ending with zucchini.

* * *


Now that the data collection stage is complete, data scientists typically use descriptive statistics and visualization techniques to better understand the data and get acquainted with it. Data scientists, essentially, explore the data to:

-   understand its content,
-   assess its quality,
-   discover any interesting preliminary insights, and,
-   determine whether additional data is necessary to fill any gaps in the data.
